### PR TITLE
Feature/Cards

### LINKED ISSUE #225: [S14.1]: Carousel
### Link to Epic

Epic: https://github.com/Open-Study-College/osc/issues/171

### Description

n/a

### Linked Tasks

- [x] #226 
- [x] #227

### LINKED ISSUE #247: [S17.1]: Cards
### Link to Epic

Epic: https://github.com/Open-Study-College/osc/issues/171

### Description

n/a

### Linked Tasks

- #249 
- #250 
- #595 
- #663

### LINKED ISSUE #249: Build Card Component
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/247

### Description

Image
Title
Copy
CTA
Badge

https://chakra-templates.dev/components/cards

Add testing
E2E
Integration

### LINKED ISSUE #250: Card: Make dynamic with Sanity
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/247

### Description

n/a

### PR DESCRIPTION
**Pull request by aaacafe786**
Closes #247 
Closes #249 
Closes #250 
Closes #225 

## 📝 Description

Refactored the carousel component to accept templates, created a basic card component

## ⛳️ Current behavior (updates)

the carousel only accepted an array of images, now you can pass any template as a child to the carousel, 

please keep in mind that there are a few styles in the carousel component that are required for it to function mostly display: flex and height and width 100%, be careful if you ever need to change this stuff in carousel.scss, you should not need to usually

when testing check out the storybook, kitchen-sink, tests and studio schema

please note, when testing controls in storybook, sometimes you must refresh the page for the changes to occur

## 🚀 New behavior

> Please describe the behavior or changes this PR adds

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: packages/osc-ui/src/components/Card/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const { className, fontSize, theme = 'primary', variant, children } = props;
+    const fontSizeClass = useFontSize(fontSize);
+    const variantClass = useVariant('c-card', variant);
+    const themeClass = useTheme(theme);
+    const classes = classNames('c-card', variantClass, fontSizeClass, themeClass, className);
+
+    return <div className={classes}>{children}</div>;
+};
+
+export const CardHeader = (props) => {
```
Comment (stevenbister) [inline packages/osc-ui/src/components/Card/Card.tsx:28]: Missing some typescript stuff

File: packages/osc-ui/src/components/Card/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    );
+
+    return (
+        <>
+            <div className={classes}>{children}</div>
+        </>
+    );
+};
+
+export const CardBody = (props) => {
```
Comment (stevenbister) [inline packages/osc-ui/src/components/Card/Card.tsx:48]: Missing some typescript stuff

File: packages/osc-ui/src/components/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const { className, fontSize, theme = 'primary', variant, children } = props;
+    const fontSizeClass = useFontSize(fontSize);
+    const variantClass = useVariant('c-card-body', variant);
+    const themeClass = useTheme(theme);
+    const classes = classNames('c-card-body', variantClass, fontSizeClass, themeClass, className);
+
+    return <div className={classes}>{children}</div>;
+};
+
+export const CardFooter = (props) => {
```
Comment (stevenbister) [inline packages/osc-ui/src/components/Card/Card.tsx:58]: Missing some typescript stuff

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx [hunk 4]
```diff
@@ -1,74 +1,160 @@
 import type { Meta, Story } from '@storybook/react';
 import React from 'react';
+import { Card, CardBody, CardHeader } from '../Card/Card';
 import type { Props } from './Carousel';
 import { Carousel } from './Carousel';
 
 export default {
-    title: 'osc-ui/Carousel',
+    title: 'osc-ui/Card Carousel',
```
Comment (stevenbister) [inline packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:8]: I think this should still be called just Carousel, perhaps we should add some stories to showcase it with different content? Cards/just images etc.

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx [hunk 5]
```diff
@@ -81,11 +167,20 @@ HorizontalOneSlide.args = {
...
-VerticalTwoSlide.args = {
+export const HorizontalThreeSlide = Template.bind({});
+HorizontalThreeSlide.args = {
+    ...Primary.args,
+    axis: 'x',
+    slidesPerPage: 3,
+    startIndex: 14
+};
+
+export const VerticalOneSlide = Template.bind({});
```
Comment (stevenbister) [inline packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:178]: Are we keeping vertical carousel? I thought we weren't gonna be using it?

File: packages/osc-ui/src/components/Carousel/carousel.scss [hunk 6]
```diff
@@ -2,100 +2,113 @@
...
+    width: 100%;
     overflow: hidden;
     display: flex;
     flex-direction: row;
     position: relative;
 
-    .embla__slide_caption {
-        padding: 0;
-        margin: 0;
+    .embla__viewport {
```
Comment (stevenbister) [inline packages/osc-ui/src/components/Carousel/carousel.scss:22]: Are these `.embla` classnames generated by embla or can we rename them something like `c-carousel__viewport` etc?

File: packages/osc-ui/src/index.tsx [hunk 7]
```diff
@@ -15,3 +15,4 @@ export { Switch } from './components/Switch/Switch';
 export { Tabs } from './components/Tabs/Tabs';
 export { Tag } from './components/Tag/Tag';
 export { Trustpilot } from './components/Trustpilot/Trustpilot';
+export { Card, CardHeader, CardBody, CardFooter } from './components/Card/Card';
```
Comment (stevenbister) [inline packages/osc-ui/src/index.tsx:18]: Can we put this in alphabetical order please?

If you set this in vscode https://discord.com/channels/915621085872029817/916253090552881153/1040286500518117416 it'll automatically do it when you save :)

**Review by stevenbister — CHANGES_REQUESTED** [review]
Hey man, looks good! A few comments and questions for ya

File: packages/osc-ecommerce/app/components/Cards/Cards.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+interface Props {
+    module: cardModule;
+}
+
+export const CardModule = (props: Props) => {
+    const { module } = props;
+
+    return (
+        <Carousel
+            active={module.carouselOptions.active} // fine
```
[the remark this answers is not in the document]
Comment (aaacafe786) [reply packages/osc-ecommerce/app/components/Cards/Cards.tsx:13]: resolved

File: packages/osc-ecommerce/app/components/Module.tsx [hunk 9]
```diff
@@ -42,6 +47,15 @@ export default function Module(props: Props) {
     const { module } = props;
 
     switch (module._type) {
+        case 'module.card':
+            const moduleCard = module as cardModule;
+            if (moduleCard.layout === 'carousel') {
+                return <CardModule module={moduleCard} />;
+            }
+            if (moduleCard.layout === 'grid') {
```
[the remark this answers is not in the document]
Comment (aaacafe786) [reply packages/osc-ecommerce/app/components/Module.tsx:55]: resolved

File: packages/osc-ui/src/index.tsx (hunk 7 above)
Comment (aaacafe786) [reply packages/osc-ui/src/index.tsx:18]: resolved and nice tip

File: packages/osc-ui/src/components/Carousel/carousel.scss (hunk 6 above)
Comment (hi-im-si) [reply packages/osc-ui/src/components/Carousel/carousel.scss:22]: agree. we should be sub-classing everything and avoiding the embla class names

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx (hunk 5 above)
Comment (hi-im-si) [reply packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:178]: I think we need to ditch it 👍🏼

File: packages/osc-ui/src/components/Card/card.scss [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        &--outline {
+            background-color: transparent;
+            border: 1px solid var(--colour-primary);
+            color: var(--colour-primary);
+        }
+    }
+
+    .c-card-body {
```
Comment (hi-im-si) [inline packages/osc-ui/src/components/Card/card.scss:46]:
```suggestion
    &__body {
```

File: packages/osc-ui/src/components/Card/card.scss [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        &--outline {
+            background-color: transparent;
+            border: 1px solid var(--colour-primary);
+            color: var(--colour-primary);
+        }
+    }
+
+    .c-card-footer {
```
Comment (hi-im-si) [inline packages/osc-ui/src/components/Card/card.scss:62]:
```suggestion
    &__footer {
```

File: packages/osc-ui/src/components/Card/card.scss [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+    &--subtle {
+        background-color: var(--colour-primary);
+        color: var(--colour-tertiary);
+    }
+
+    &--outline {
+        background-color: transparent;
+        border: 1px solid var(--colour-primary);
+        color: var(--colour-primary);
```
[the remark this answers is not in the document]
Comment (hi-im-si) [reply packages/osc-ui/src/components/Card/card.scss:23]: i think until we see what the cards actually look like visually then we should keep this file very basic

File: packages/osc-ui/src/components/Card/Card.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+            justifyContent: 'center',
+            height: '100%'
+        }}
+    >
+        <Card>
+            <CardHeader>
+                <img
+                    style={{
+                        width: '100%'
+                    }}
```
Comment (hi-im-si) [inline packages/osc-ui/src/components/Card/Card.stories.tsx:25]: Is there a specific reason for choosing to use inline styles of the main card.scss?

File: packages/osc-ecommerce/app/components/Module.tsx [hunk 14]
```diff
@@ -42,6 +47,15 @@ export default function Module(props: Props) {
...
+                console.log(moduleCard);
```
Comment (hi-im-si) [inline packages/osc-ecommerce/app/components/Module.tsx:56]: remove `console.log`

**Review by hi-im-si — COMMENTED** [review]
Aside from the comments, having reviewed the [deploy link](https://pr-540-osc-ecommerce.fly.dev/kitchen-sink) I am unable to see the Cards working with the Grid layout - selecting this option does not output any cards in the DOM

**Review by hi-im-si — CHANGES_REQUESTED** [review]
see previous comments

File: packages/osc-ecommerce/app/root.tsx [hunk 15]
```diff
@@ -26,6 +27,7 @@ let isMount = true;
 export const links: LinksFunction = () => {
     return [
         { rel: 'stylesheet', href: oscUiCarouselStyles },
+        { rel: 'stylesheet', href: oscUiCardStyles },
```
[the remark this answers is not in the document]
Comment (aaacafe786) [reply packages/osc-ecommerce/app/root.tsx:30]: done

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx (hunk 4 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:8]: done

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx (hunk 5 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:178]: ive disabled the vertical option for now

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx (hunk 4 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:8]: unfortunately, i think we need the classnames for the carousel to function

File: packages/osc-ui/src/components/Carousel/carousel.scss (hunk 6 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Carousel/carousel.scss:22]: i think we need embla classes to function, eg embla__slide

File: packages/osc-ui/src/components/Card/card.scss (hunk 10 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Card/card.scss:46]: done

File: packages/osc-ui/src/components/Card/card.scss (hunk 11 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Card/card.scss:62]: done

File: packages/osc-ui/src/components/Card/Card.stories.tsx (hunk 13 above)
Comment (aaacafe786) [reply packages/osc-ui/src/components/Card/Card.stories.tsx:25]: its just a container to hold the card, not part of the actual card component

File: packages/osc-ecommerce/app/components/Module.tsx (hunk 14 above)
Comment (aaacafe786) [reply packages/osc-ecommerce/app/components/Module.tsx:56]: done

File: packages/osc-ui/src/components/Carousel/carousel.scss (hunk 6 above)
Comment (stevenbister) [reply packages/osc-ui/src/components/Carousel/carousel.scss:22]: I think it's fine to keep the classes in the markup (and maybe add a comment as to why they're there) but maybe add the `c-carousel` subclasses to handle the styling

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx (hunk 4 above)
Comment (stevenbister) [reply packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:8]: Don't think this is done mate?

File: packages/osc-ui/src/components/Carousel/Carousel.stories.tsx (hunk 5 above)
Comment (stevenbister) [reply packages/osc-ui/src/components/Carousel/Carousel.stories.tsx:178]: Do you mean it's been disabled in Studio? Can we take it out of storybook as well?

File: packages/osc-ecommerce/app/components/Cards/Cards.tsx [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+                                              >
+                                                  {tag}
+                                              </span>
+                                          );
+                                      })}
+                                  </div>
+                                  <p style={{ color: 'white' }}>{item.title}</p>
+                                  <p>
+                                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam
+                                      quia quo iure vero. Blanditiis error perspiciatis aperiam
```
[the remark this answers is not in the document]
Comment (aaacafe786) [reply packages/osc-ecommerce/app/components/Cards/Cards.tsx:65]: done

**Comment by hi-im-si** [comment]
Referring to @aaacafe786 confirmation that this PR can be closed without merging

**Pull request closed without merging**